File: Project Euler #18/CPP/solution18.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin>>t; 
    while(t--){
        int n;
        cin>>n;
        int arr[n][n];
        int j =0;
        for(int i=0;i<n;i++){
            for(int k=0;k<=j;k++){
                cin>>arr[i][k];
            }
            j++;
        }
        for(int i=n-2;i>=0;i--){
            for(int j = 0; j <= i; j++){
                arr[i][j] += max(arr[i + 1][j], arr[i + 1][j + 1]); 
            }
        }
        cout<<arr[0][0]<<endl;
    } 
    return 0;
}

File: Project Euler #18/Python/solution18.py
if __name__ == '__main__':
    t = int(input())
    for a in range(0, t):
        n = int(input())
        arr = list()
        for i in range(n):
            temp = list(map(int, input().split()))
            arr.append(temp)
        j = 0
        for i in range(n-2, -1, -1):
            for j in range(0, i+1):
                arr[i][j] += max(arr[i+1][j], arr[i+1][j+1])   
        print(arr[0][0])        
